modules/data_models.py

```diff
@@ -0,0 +1,363 @@
+            None
+        """
+        self.highest_temperature = get_highest_temperature(self.line)
+        self.lowest_temperature = get_lowest_temperature(self.line)
+        self.max_humidity = get_max_humidity(self.line)
+        self.mean_humidity = get_mean_humidity(self.line)
+        self.date = get_date(self.line)
+
+    def get_max_temperature(self):
+        """
+        Returns the highest_temperature, can be None
+        Returns:
+            (int or None):  highest_temperature in degree celsius
+                            Or
+                            None
+        """
+
+        return self.highest_temperature
+
+    def get_min_temperature(self):
+        """
+        Returns the lowest_temperature, can be None
+        Returns:
+            (int or None):  lowest_temperature in degree celsius
+                            Or
+                            None
+        """
+
+        return self.lowest_temperature
+
+    def get_max_humidity(self):
+        """
+        Returns the max_humidity, can be None
+        Returns:
+            (int or None):  max_humidity in percentage
+                            Or
+                            None
+        """
+
+        return self.max_humidity
+
+    def get_mean_humidity(self):
+        """
+        Returns the mean_humidity, can be None
+        Returns:
+            (int or None):  mean_humidity in percentage
+                            Or
+                            None
+        """
+
+        return self.mean_humidity
+
+    def get_date(self):
+        """
+        Returns the date
+        Returns:
+            (str): Value containing date in yyyy-mm-dd format e.g '2005-6-2'
+        """
+
+        return self.date
+
+
+class MonthData:
+    """
+    This class holds the data of an entire month
+    """
+
+    def __init__(self, year, month, path=WEATHER_FILES_DIR):
+        """
+        Initializes the members name, year and path and
+        calls populate to fill days_data list
+        Args:
+            year(str or int):   Value containing 4 digit year e.g '2004'
+            month(str): Value containing 3 character month name e.g 'Feb', 'Mar'
+            path(str):  Value containing path to weather files e.g 'weatherfiles/'
+        """
+        self.days_data = []
+        self.name = month
+        self.year = year
+        self.path = path
+        self.populate()
+
+    def populate(self):
+        """
+        Initializes the days_data list with DayData objects for each day of the month
+        Returns:
+            None
+        """
+
+        month_data = next(read_data(f"{self.year}_{self.name[0:3]}", self.path))
+        for line in month_data:
+            day_data = DayData(parse_line(line))
+            self.days_data.append(day_data)
+
+    def get_max_extremes_with_date(self):
+        """
+        Returns the max highest temperature, min lowest temperature and max humidity
+        for the whole month with respective dates
+        Returns:
```

Comment:
OK, how about we make this scalable. e.g what if we only want to get the max temperature of the month? We will need to call this method and then fetch a value from it which will be an extensive operation in this case. Now that we have the list of DayDaya objects we should divide this function into other independent functions e.g. `get_max_month_temperature`, `get_min_month_temperature` and others.

Reply:
Done